conda_testenv/test_env.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import print_function
+
+from contextlib import contextmanager
+import os
+import shutil
+import tempfile
+
+import conda.cli.main_list
+import conda.install as install
+from conda_build.config import config
+import conda_build.metadata
+from conda_build.scripts import prepend_bin_path
+
+from conda_testenv import conda_build_test
+
+
+def list_package_sources(prefix):
+    """
+    List the sources of all the packages installed in the given environment.
+
+    """
+    installed = install.linked(prefix)
+    sources = []
+    for dist in conda.cli.main_list.get_packages(installed, None):
+        info = install.is_linked(prefix, dist)
+        sources.append(info['link']['source'])
+    return sources
+
+
+def recipe_exists(source):
```

Comment:
The name suggests a boolean return value, but that isn't the case.

Stylistically, I wonder if you are better going down the "ask forgiveness" approach:

```
def recipe_directory(distribution_source):
    if exists: return path
    else: raise IOError (or a custom error, or something)
```

and then dealing with that in your code.